### PR TITLE
ripe v3 dao fork front end

### PR DESCRIPTION
**Pull request by ripe0x**
# Change summary

1. **Proposal editing**: UI updates to allow proposers to update proposals while within the updatablePeriod. UI updates include the addition of a "Version X" element below the title, proposal editing view, a updatable proposal pill, and a new proposal history page for each proposal version. If a proposal is updatable, the proposer will see an `Edit` button on the proposal view alongside the `Cancel` button. 
  Examples: 
    - [Edited proposal](https://ripe-v3-dao-fork-front-end--charming-parfait-9b3b08.netlify.app/vote/10)
    - [Proposal history](https://ripe-v3-dao-fork-front-end--charming-parfait-9b3b08.netlify.app/vote/10/history/)
    - [Proposal version permalink](https://ripe-v3-dao-fork-front-end--imaginative-arithmetic-48ce04.netlify.app/vote/14/history/2)
2. **Propose by signature**: Referenced as candidates in the UI. Anyone can submit a proposal candidate (non-voters pay a fee), which can then be sponsored by nouns voters. Like proposals, candidates are also editable. UI updates include candidate proposal view, candidate editing view candidate history, candidate sponsoring flow, candidate proposal flow.
  Examples: 
    - [Candidate](http://localhost:3000/candidates/0xcb43078c32423f5348cab5885911c3b5fae217f9-first-candidate)
    - [Candidate history](http://localhost:3000/candidates/0x57a39aa135a688cf18ece526f1d3597a11e1b32a-candidate-to-proposal-flow/history/)
    - Sponsored votes with messages
image
3. **Feedback on proposals and candidates**: nouns voters can submit written feedback and a voting direction on updatable proposals and proposal candidates. UI updates include adding a sidebar to proposals to hold the feedback component, creating states for viewing and submitting feedback onchain.
  Examples: 
    - [Candidate with feedback](http://localhost:3000/candidates/0xcb43078c32423f5348cab5885911c3b5fae217f9-first-candidate)
image
4. **Objection-only period**: as described in #706, UI updates include the addition of a new alert bar during the period. once the objection period has been triggered, but before the period starts, a short description appears below the `Ends` element to give additional context. 
Examples (using screenshots because the state is time-based): 
image
5. **Votes Snapshot After Voting Delay**: as described in #706. 
6. **Nouns Fork**: UI additions include a new `/forks` page that lists all fork instances, and a `/fork/[id]` page, add nouns to escrow/fork with reason, withdraw from escrow, and empty, escrow, forking, forked states.
  Examples: 
    - [Forks page](https://ripe-v3-dao-fork-front-end--charming-parfait-9b3b08.netlify.app/fork)
    - [Fork in escrow](https://ripe-v3-dao-fork-front-end--charming-parfait-9b3b08.netlify.app/fork/0)
    - Forking (will update link when we have an active example state on staging)
    - [Forked](https://ripe-v3-dao-fork-front-end--imaginative-arithmetic-48ce04.netlify.app/fork/5) 
7. **Global nav**: added a dropdown to the DAO nav item that exposes Proposals, Candidates, and Fork. 
8. **Feature toggles**: The v3 features can be toggled on/off [using the config here](https://github.com/ripe0x/nouns-monorepo/blob/018c2bd705f62a3e978af0b6a64239f6a26c56f6/packages/nouns-webapp/src/config.ts#L150). As an extra measure, there is also a hook implemented that auto-checks to see if the site is connected to v3 contracts, and only exposes the v3 features if that's the case.

# Upgrade Plan
Once the v3 data contract has been deployed and added to subgraph, this code works with v2 and v3 contracts, with an auto-detect hook that toggles features based on the contract versions. So once the v3 contracts are updated, the UI will update automatically. If v3 features need to be hidden post-upgrade, they can be toggled off [in the config](https://github.com/ripe0x/nouns-monorepo/blob/018c2bd705f62a3e978af0b6a64239f6a26c56f6/packages/nouns-webapp/src/config.ts#L150).

### CODE REVIEW

File: packages/nouns-webapp/src/components/NavBar/index.tsx [hunk 1]
```diff
@@ -50,6 +52,46 @@ const NavBar = () => {
     : NavBarButtonStyle.WARM_INFO;
 
   const closeNav = () => setIsNavExpanded(false);
+  const buttonClasses = usePickByState(
+    navDropdownClasses.whiteInfoSelectedBottom,
+    navDropdownClasses.coolInfoSelected,
+    navDropdownClasses.warmInfoSelected,
+    history,
+  );
+  const candidatesNavItam = config.featureToggles.candidates ? (
```
Comment (eladmallel) [inline packages/nouns-webapp/src/components/NavBar/index.tsx:61]: typo Itam -> Item

Comment (ripe0x) [reply packages/nouns-webapp/src/components/NavBar/index.tsx]: thanks. resolved.

File: netlify.toml [hunk 2]
```diff
@@ -3,13 +3,17 @@
   REACT_APP_ENABLE_HISTORY = "true"
 
 [context."staging".environment]
-  REACT_APP_CHAIN_ID = "1"
+  REACT_APP_CHAIN_ID = "11155111"
```
Comment (solimander) [inline netlify.toml:6]: Staging should point at mainnet

File: packages/nouns-webapp/src/wrappers/nounsDao.ts [hunk 3]
```diff
@@ -646,3 +929,430 @@ export const useExecuteProposal = () => {
...
+export const useIsDaoGteV3 = (): boolean => {
+  const [implementation] =
+    useContractCall({
+      abi: abiV2,
+      address: config.addresses.nounsDAOProxy,
+      method: 'implementation',
+    }) || [];
+  if (
+    implementation &&
+    config.addresses.nounsDAOLogicV2 &&
```
Comment (eladmallel) [inline packages/nouns-webapp/src/wrappers/nounsDao.ts:1329]: @ripe0x heads up, this address in `addresses.json` is NOT the latest logic we're using on mainnet is `0x51C7D7C47E440d937208bD987140D6db6B1E4051`

File: packages/nouns-webapp/src/wrappers/nounsDao.ts [hunk 4]
```diff
@@ -646,3 +929,430 @@ export const useExecuteProposal = () => {
...
+    nounsDaoContract,
+    'executeFork',
+  );
+  return { executeFork, executeForkState };
+};
+
+export const useGracePeriod = () => {
+  const [gracePeriod] =
+    useContractCall<[any]>({
+      executorAbi,
```
Comment (davidbrai) [inline packages/nouns-webapp/src/wrappers/nounsDao.ts:1223]: @ripe0x this should be `abi: executorAbi`, but looks like this code is unused, so maybe just delete it?

File: packages/nouns-sdk/src/contract/addresses.json [hunk 5]
```diff
@@ -34,7 +34,7 @@
     "nounsAuctionHouseProxyAdmin": "0x5dAdD045E2aC09f55f49Bb84886Ea7724c675a0d",
     "nounsDaoExecutor": "0xc15008dE43D93D115BD64ED4D95817fFdBfb6DEA",
     "nounsDAOProxy": "0x22F7658f64be277e6b3968ecE7b773b092a39864",
-    "nounsDAOLogicV2": "0x3c793c2D7Cfd48046c75c63E43d36a2cB42fC649",
```
Comment (davidbrai) [inline packages/nouns-sdk/src/contract/addresses.json]: @ripe0x why did you need to remove it?

File: packages/nouns-contracts/tasks/deploy-local-dao-v3.ts [hunk 6]
```diff
@@ -50,7 +50,13 @@ task('deploy-local-dao-v3', 'Deploy contracts to hardhat')
...
   .addOptionalParam('timelockDelay', 'The timelock delay (seconds)', 60 * 60 * 24 * 2, types.int) // Default: 2 days
   .addOptionalParam('votingPeriod', 'The voting period (blocks)', 4 * 60 * 24 * 3, types.int) // Default: 3 days
-  .addOptionalParam('votingDelay', 'The voting delay (blocks)', 1, types.int) // Default: 1 block
+  .addOptionalParam('votingDelay', 'The voting delay (blocks)', 100, types.int) // Default: 1 block
+  .addOptionalParam(
+    'proposalUpdatablePeriodInBlocks',
+    'The updatable period in blocks',
+    100,
+    types.int,
+  ) // Default: 1 block
```
Comment (eladmallel) [inline packages/nouns-contracts/tasks/deploy-local-dao-v3.ts:59]: inline comment says 1 block but it's now 100 blocks.
also thinking we should make this short again, maybe 10 blocks, wdyt?

File: packages/nouns-webapp/src/components/CandidateSponsors/SignatureForm.tsx [hunk 7]
```diff
@@ -0,0 +1,448 @@
...
+  //   const today = new Date();
+  //   const date = +dayjs(e.target.value);
+  //   if (+dayjs(e.target.value) > +dayjs(today)) {
+  //     // e.target.value = today.toISOString().split('T')[0];
+  //     // setExpirationDate(+dayjs(e.target.value));
+  //     setDateErrorMessage('');
+  //   } else {
+  //     setDateErrorMessage('Date must be in the future');
+  //   }
+  // };
```
Comment (solimander) [inline packages/nouns-webapp/src/components/CandidateSponsors/SignatureForm.tsx:275]: Remove

File: packages/nouns-sdk/src/contract/addresses.json
Comment (eladmallel) [inline packages/nouns-sdk/src/contract/addresses.json:1]: let's make sure to update mainnet's `nounsDAOLogicV2` to the latest value from mainnet

File: packages/nouns-webapp/src/hooks/useTreasuryBalance.ts [hunk 8]
```diff
@@ -12,7 +12,7 @@ import { BigNumber, ethers } from 'ethers';
  */
 export const useTreasuryBalance = () => {
   const ethBalance = useEtherBalance(config.addresses.nounsDaoExecutor);
-  const lidoBalanceAsETH = useLidoBalance();
+  const lidoBalanceAsETH = useLidoBalance(config.addresses.nounsDaoExecutor);
```
Comment (eladmallel) [inline packages/nouns-webapp/src/hooks/useTreasuryBalance.ts:15]: NTH: add rocket ETH balance as well

File: packages/nouns-webapp/src/pages/Fork/DeployForkButton.tsx [hunk 9]
```diff
@@ -0,0 +1,171 @@
...
+        )}
+        {errorMessage && errorMessage}
+      </p>
+    </div>
+  );
+
+  return (
+    <>
+      {!props.isForkPeriodActive && props.isThresholdMet && (
+        <button
```
Comment (solimander) [inline packages/nouns-webapp/src/pages/Fork/DeployForkButton.tsx:140]: Button should be disabled when no account is connected

File: packages/nouns-sdk/src/contract/addresses.json (hunk 5 above)
Comment (ripe0x) [reply packages/nouns-sdk/src/contract/addresses.json]: i think it got removed when i was testing the v2 contracts. added back in.

File: packages/nouns-contracts/tasks/deploy-local-dao-v3.ts (hunk 6 above)
Comment (ripe0x) [reply packages/nouns-contracts/tasks/deploy-local-dao-v3.ts:59]: i needed a longer window for testing the updatable state. but it's easy to switch back.

File: packages/nouns-webapp/src/hooks/useTreasuryBalance.ts (hunk 8 above)
Comment (ripe0x) [reply packages/nouns-webapp/src/hooks/useTreasuryBalance.ts:15]: can you say more about this?

File: packages/nouns-webapp/src/pages/Fork/DeployForkButton.tsx (hunk 9 above)
Comment (ripe0x) [reply packages/nouns-webapp/src/pages/Fork/DeployForkButton.tsx:141]: yep yep. good catch. updated here https://github.com/nounsDAO/nouns-monorepo/pull/745/commits/95c7cf24c95f79f99dae0b524cc7e88e737eb18f

File: packages/nouns-sdk/src/contract/addresses.json
Comment (ripe0x) [reply packages/nouns-sdk/src/contract/addresses.json:1]: resolved here https://github.com/nounsDAO/nouns-monorepo/pull/745/commits/394b412e6997b738a881ca8ff0a93af95e74cd3a

File: packages/nouns-contracts/tasks/deploy-local-dao-v3.ts (hunk 6 above)
Comment (ripe0x) [reply packages/nouns-contracts/tasks/deploy-local-dao-v3.ts]: resolved here: https://github.com/nounsDAO/nouns-monorepo/pull/745/commits/2419f27ace2558034ed2291b05cc597f04dab2e2

File: packages/nouns-webapp/src/hooks/useTreasuryBalance.ts (hunk 8 above)
Comment (ripe0x) [reply packages/nouns-webapp/src/hooks/useTreasuryBalance.ts:15]: closing this out per our chat in discord

**Review by solimander — APPROVED** [review]
